app/api/business/team_business.py

```diff
@@ -0,0 +1,309 @@
+
+
+def create_team():
+    user = users.get(current_user.id)
+    created_teams = team_service.get_teams_for_user(user.id, 'created')
+    completed_teams = team_service.get_teams_for_user(user.id)
+
+    new_team = None
+    if len(completed_teams) == 0:
+        if len(created_teams) == 0:
+            team = team_service.save(
+                Team(
+                    name='',
+                    status='created',
+                    team_members=[
+                        TeamMember(
+                            user_id=user.id,
+                            is_team_lead=True
+                        )
+                    ]
+                )
+            )
+
+            audit_service.log_audit_event(
+                audit_type=audit_types.create_team,
+                db_object=team,
+                user=current_user.email_address
+            )
+
+            publish_tasks.team.delay(
+                publish_tasks.compress_team(team),
+                'created'
+            )
+
+            return get_team(team.id)
+
+        created_team = created_teams.pop()
+        return get_team(created_team.id)
+    else:
+        team = completed_teams[0]
+        raise TeamError('You can only be in one team. You\'re already a member of {}.'.format(team.name))
+
+
+def get_teams_overview():
+    teams_overview = {}
+    user = users.get(current_user.id)
+    completed_teams = team_service.get_teams_for_user(user.id)
+    email_domain = get_email_domain(user.email_address)
+
+    overview = team_service.get_teams_overview(user.id, email_domain)
+    teams_overview.update(overview=overview)
+
+    organisation = users.get_user_organisation(email_domain)
+    teams_overview.update(organisation=organisation)
+
+    completed_teams_count = len(completed_teams)
+    teams_overview.update(completedTeamsCount=completed_teams_count)
+
+    return teams_overview
+
+
+def get_people_overview():
+    people_overview = {}
+    user = users.get(current_user.id)
+    completed_teams = team_service.get_teams_for_user(user.id)
+    domain = get_email_domain(user.email_address)
+
+    people = users.get_buyer_team_members(domain)
+    people_overview.update(users=people)
+
+    organisation = users.get_user_organisation(domain)
+    people_overview.update(organisation=organisation)
+
+    completed_teams_count = len(completed_teams)
+    people_overview.update(completedTeamsCount=completed_teams_count)
+
+    return people_overview
+
+
+def get_team(team_id):
+    team_member = team_member_service.find(
+        is_team_lead=True,
+        team_id=team_id,
+        user_id=current_user.id
+    ).one_or_none()
+
+    if not team_member:
+        raise UnauthorisedError('Only team leads can edit a team')
+
+    team = team_service.find(id=team_id).one_or_none()
+    if not team:
+        raise NotFoundError('{} does not exist'.format(team_id))
+
+    domain = get_email_domain(current_user.email_address)
+    team = team_service.get_team(team_id)
```

Comment:
`team` is already established above on L107, although that is using the simpler `.find()`. Do we need to use `.get_team()` here? if so, can it replace L107?

Reply:
fixed